src/Components/Postulants/Form/index.js

```diff
@@ -0,0 +1,714 @@
+    const primaryStudies = {
+      startDate: primarySDValue,
+      endDate: primaryEDValue,
+      school: primarySchoolValue
+    };
+
+    const secondaryStudies = {
+      startDate: secondarySDValue,
+      endDate: secondaryEDValue,
+      school: secondarySchoolValue
+    };
+
+    const tertiaryStudies = [
+      {
+        startDate: tertiarySDValue,
+        endDate: tertiaryEDValue,
+        description: tertiaryDescriptionValue,
+        institute: tertiaryInstituteValue
+      }
+    ];
+
+    const universityStudies = [
+      {
+        startDate: universitySDValue,
+        endDate: universityEDValue,
+        description: universityDescriptionValue,
+        institute: universityInstituteValue
+      }
+    ];
+
+    const informalStudies = [
+      {
+        startDate: informalSDValue,
+        endDate: informalEDValue,
+        description: informalDescriptionValue,
+        institute: informalInstituteValue
+      }
+    ];
+
+    return {
+      primaryStudies,
+      secondaryStudies,
+      tertiaryStudies,
+      universityStudies,
+      informalStudies
+    };
+  };
+
+  const workExperienceBodyConstructor = () => {
+    const workExperience = [
+      {
+        company: workExperienceCompanyValue,
+        startDate: workExperienceSDValue,
+        endDate: workExperienceEDValue,
+        description: workExperienceDescriptionValue
+      }
+    ];
+
+    return workExperience;
+  };
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    // saveButton.disables = !!params.get('id');
+
+    let url = '';
+
+    const options = {
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        firstName: firstNameValue,
+        lastName: lastNameValue,
+        email: emailValue,
+        password: passwordValue,
+        address: adressValue,
+        birthday: birthdayValue,
+        available: availableValue,
+        phone: phoneValue,
+        profiles: undefined /* profilesBodyContructor() */,
+        contactRange: {
+          from: contactFromValue.replace(':', ''),
+          to: contactToValue.replace(':', '')
+        },
+        studies: studiesBodyConstructor(),
+        workExperience: workExperienceBodyConstructor()
+      }),
+      method: 'POST'
+    };
+
+    options.method = 'POST';
+    url = `${process.env.REACT_APP_API}/postulants`;
+
+    if (postulantId) {
+      options.method = 'PUT';
+      url = `${process.env.REACT_APP_API}/postulants/${postulantId}`;
+    }
+
+    fetch(url, options)
+      .then((response) => {
+        if (response.status !== 200 && response.status !== 201) {
+          return response.json().then(({ message }) => {
+            throw new Error(message);
+          });
+        }
+        return response.json();
+      })
+      .then(() => {
+        window.location.href = `${window.location.origin}/postulants`;
+      })
+      .catch((error) => {
+        console.log(JSON.stringify(error));
```

Comment:
the user will never know what happened if we do not show it in the UI

Reply:
added `.catch()`